### ISSUE TITLE
Not using adapter root?

### ISSUE THREAD

**Issue by alanhartless** [Falc/FlysystemLocalSymlinkPlugin#5]
Not sure if this is a plugin issue, Flysystem issue, or a coding issue but it seems that the $target and $symlink passed into handle() does not include the $root when setting up the LocalAdaptor.  So, the symlink fails when the service utilizing the specific Flysystem service that is set with a root other than /.  Thus the benefit of using the plugin is lost if I have to pass the root to the service anyway when I can just use symlink().  

Thoughts? I'm hoping it's just a coding issue on my part.

**Comment by Falc** [comment]
@alanhartless can you provide any code in order to take a look and find out where the problem is?

**Comment by Falc** [comment]
@alanhartless ping!

**Comment by alanhartless** [comment]
Hi!  Sorry, yes.

Here's an example of where we define a service in our container:

```
        $container->share(
            'flysystem.files',
            function (Container $container) {
                /** @var \Joomla\Registry\Registry $config */
                $config = $container->get('config');

                return new Filesystem(new Local($config->get('file_path')));
            }
        );
```

`file_path` is something like `/var/www/html/files`.  That service is then passed into other services that need access to the defined `file_path`.

By passing a specific directory to `Local()`, all the built in Flysystem methods take arguments that are relative to that path.

But when it comes to using the symlink plugin, if I pass in relative target and symlink names, it doesn't prepend the `file_path` so it fails.

I would expect to be able to do something like this `$flysystem->symlink('user/something', 'user/something_else');`
 to symlink `/var/www/html/files/user/something_else` to `/var/www/html/files/user/something` since this is the behavior of the other Flysystem methods. But it fails because the plugin is doing `symlink('user/something',  'user/something_else')`;

Flysystem doesn't allow me to retrieve the base directory back out it's object so I have to pass `file_path` as an additional dependency to the service so that I can create a full path.  Since I have to do that, it doesn't make sense to use the plugin when I can simply use `symlink()` directly.

Does that help/make sense?

Thanks!!!
Alan

**Comment by Falc** [comment]
You are absolutely right. I wrote this plugin a lot of time ago for a simple project and got used to work with absolute paths. I have pushed [some changes](https://github.com/Falc/FlysystemLocalSymlinkPlugin/commit/b0222dffa452ea0a5fff921ecfc25a8905e3f1a4) so this should be fixed.

Install/update the plugin using "dev-master" in composer and try it. I will tag a new version if it works.

**Comment by lsv** [comment]
ping @alanhartless did it work with dev-master?
If it did, could @Falc please tag a new version? :)

**Comment by Falc** [comment]
I tested myself some days ago so I am going to update the README and tag a new version in the next days.

**Comment by lsv** [comment]
@Falc that would be so awesome :+1:

**Comment by Falc** [comment]
This is fixed in v1.2.0. :)
